app/forms/support_user_invite_form.rb

```diff
@@ -0,0 +1,54 @@
+class SupportUserInviteForm
+  FORM_PARAMS = %i[first_name last_name email].freeze
+  include ActiveModel::Model
+
+  attr_accessor :email, :first_name, :last_name, :service
+
+  validate :validate_support_user
+  validates :service, presence: true
+
+  def invite
+    return false unless valid?
+
+    save_support_user
+  end
+
+  def as_form_params
+    { "support_user" => slice(FORM_PARAMS) }
+  end
+
+  private
+
+  def support_user
+    @support_user ||=
+      begin
+        user = user_klass
+          .with_discarded
+          .discarded
+          .find_or_initialize_by(email:)
+        user.assign_attributes(first_name:, last_name:)
+        user
+      end
+  end
+
+  def save_support_user
+    ActiveRecord::Base.transaction do
+      support_user.undiscard! if support_user.discarded?
+      support_user.save!
+      SupportUser::Invite.call(support_user:)
```

Comment:
not important, but you don't actually need to put the service in the transaction. If one of the other two methods fail, it will raise an exception and the rest of the code won't be executed anyway. There's no way to 'rollback' `Invite.call`, so it feels a bit weird putting it in here. 

But not a big deal. Not a blocker.

Reply:
resolved in [2972587](https://github.com/DFE-Digital/itt-mentor-services/pull/253/commits/297258709781727ccf54edf9d0813505c2b43e42)